rfcs/0136-stabilize-incrementally.md

```diff
@@ -0,0 +1,189 @@
+- How can `nix repl` have a more normal CLI?
+
+## Step 4: Stabilize Flakes itself
+
+Finally, with the other less controversial interfaces stabilized, we can tackle the Flakes itself, the one remainder.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+Having laid out the plan, let us now return to how the current situation is characterized and see if the various facts that the factions orient themselves are respected.
+
+## Flakes are very popular
+
+And they will be stabilized, with minimal delay.
+
+Firstly, there is no feature work proposed in the interim --- the splitting of Nix is just partitioning existing functionality, with some behind-the-scenes refactors needed to make that possible.
+The deadline placed on that work also assuages those concerned about the current "limbo" that stabilization will be gotten to no matter what, even if the splitting fails, after a set amount of time.
+
+## Difficulties in the roll-out
+
+Process concerns are addressed by having a clear process, with clear outcomes, before any stabilization is begun.
+That is the purpose of this RFC!
+
+## Flakes are criticized for eating other Nix features
+
+We do *not* propose adding delay to once-again separate those features from Flakes, as that would introduce more feature work which would delay stabilization and be unacceptable to the pro-Flakes faction.
+
+That said, an ancillary incremental process of incremental stabilization is to bolster a *sense* of layering in Nix's user interface that has been, according to this camp, lost.
+Ensuring that the these two subsets of the new CLI --- without Flakes, and without Nix language support --- do in fact make sense in isolation will provide a "scaffolding" upon which interested parties can later introduce generalized features like search and pure eval without Flakes.
+
+The hope is that such scaffolding will assuage this faction their concerns are heard without holding things up.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The main downside is a small delay from the splitting Nix process, and then delay between the stabilization steps.
+
+The first delay has a guaranteed upper bound, thanks to its self-imposed deadline.
+That maximum delay should be dwarfed by duration of time we've spent "in limbo" without a clear plan to move forward.
+I therefore think that is a small and reasonable price to pay for the benefit of community harmony.
+
+The second delays don't represent time "ideal" from a stabilization perspective.
+As long as we are making progress stabilizing features and having healthy discussions, I don't see any problem.
+
+# Alternatives
+[alternatives]: #alternatives
+
+Of course, we could instead count on the magnitude of new years making the concerns of old users unimportant, but I think the plan here has little downsides; we can instead make everyone happy with only a small delay.
+If that is true, why not do that instead!
+A plan to stabilize F
```

Comment:
Did something go missing here?

Reply:
I think I actually just didn't finish the thought, whoops!

Reply:
```suggestion
```